tybalt/models.py

```diff
@@ -185,13 +115,138 @@ def connect_layers(self):
+        self.learning_rate = learning_rate
+        self.kappa = kappa
+        self.epsilon_std = epsilon_std
+        self.beta = beta
+        self.loss = loss
+
+    def build_encoder_layer(self):
+        """
+        Function to build the encoder layer connections for conditional VAE
+        """
+        # Input place holder for RNAseq and label data with specific input size
+        self.rnaseq_input = Input(shape=(self.original_dim, ))
+        self.label_input = Input(shape=(self.label_dim, ))
+
+        # Concatenate input layers to obtain single input into model
+        self.cvae_input = concatenate([self.rnaseq_input, self.label_input])
+
+        # Input layer is compressed into a mean and log variance vector of
+        # size `latent_dim`. Each layer is initialized with glorot uniform
+        # weights and each step (dense connections, batch norm, and relu
+        # activation) are funneled separately.
+        # Each vector are connected to the rnaseq input and label tensors
+
+        # input layer to latent mean layer
+        z_mean = Dense(self.latent_dim,
+                       kernel_initializer='glorot_uniform')(self.cvae_input)
+        z_mean_batchnorm = BatchNormalization()(z_mean)
+        self.z_mean_encoded = Activation('relu')(z_mean_batchnorm)
+
+        # input layer to latent standard deviation layer
+        z_var = Dense(self.latent_dim,
+                      kernel_initializer='glorot_uniform')(self.cvae_input)
+        z_var_batchnorm = BatchNormalization()(z_var)
+        self.z_var_encoded = Activation('relu')(z_var_batchnorm)
+
+        # return the encoded and randomly sampled z vector
+        # Takes two keras layers as input to the custom sampling function layer
+        self.z = Lambda(self._sampling,
+                        output_shape=(self.latent_dim, ))([self.z_mean_encoded,
+                                                           self.z_var_encoded])
+
+        # To make the model conditional, add back the label layer to the latent
+        # features. This will encourage the z vector to learn common sources of
+        # variation, while the label input layer can extract variation
+        # conditioned on the specific sample labels.
+        self.zc = concatenate([self.z, self.label_input])
+
+    def build_decoder_layer(self):
+        """
+        Function to build the decoder layer connections for conditional VAE
+        """
+        # The decoding layer is much simpler with a single layer glorot uniform
+        # initialized and sigmoid activation
+
+        self.cvae_input_dim = self.original_dim + self.label_dim
+        self.cvae_latent_dim = self.latent_dim + self.label_dim
+
+        self.decoder_model = Sequential()
+        self.decoder_model.add(Dense(self.cvae_input_dim, activation='sigmoid',
+                                     input_dim=self.cvae_latent_dim))
+        self.rnaseq_reconstruct = self.decoder_model(self.zc)
+
+    def compile_vae(self):
```

Comment:
I think it could be private for now to reduce complexity. But I may want to convert it back to public in the future. See [this comment above](https://github.com/greenelab/tybalt/pull/105#discussion_r167052996)